### PR TITLE
Refactor TITLE_SCRIPT to use strong enum

### LINKED ISSUE #12443: Refactor TITLE_SCRIPT to use strong enum

### PR DESCRIPTION
**Pull request by mysterious1223**
Update the enum TITLE_SCRIPT in [TitleSequence.h](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/title/TitleSequence.h) & [TitleSequence.cpp](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/title/TitleSequence.cpp) to use enum class. Closes #12443

### CODE REVIEW

**Comment by James103** [comment]
Your commit message is too short. Could you please use something like "Fix #12443: Refactor TITLE_SCRIPT to use strong enum"?

**Comment by duncanspumpkin** [comment]
@James103 this is just a placeholder for now to show its being worked on.

**Comment by mysterious1223** [comment]
Yes still working on this. I can submit another one if you need me to? and create a commit called "Fix #12443: Refactor TITLE_SCRIPT to use strong enum".

However, I will create a actual pull request when this is completed.

**Comment by duncanspumpkin** [comment]
In general keep to a singular PR. Its easy enough to reword commit messages. If you don't know then I can guide you using `git rebase -i`

**Comment by mysterious1223** [comment]
I think this should be complete. Please let me know if I should make any changes. Thanks.

**Comment by duncanspumpkin** [comment]
Can you rebase this to remove the merge commits.

**Comment by mysterious1223** [comment]
I was able to remove them.

**Comment by mysterious1223** [comment]
Looks like my changes didn't go through. Sorry about that, not sure what went wrong. I will convert to draft.

**Comment by mysterious1223** [comment]
Alright. Commit should be good now. 
Still trying to get use to this!

**Review by tupaschoal — COMMENTED** [review]
There is still a merge conflict, try rebasing and solving it. There's a tutorial here:
https://github.com/OpenRCT2/OpenRCT2/wiki/Rebase-and-Sync-fork-with-OpenRCT2

**Comment by mysterious1223** [comment]
Thanks! I'll try that.

**Comment by mysterious1223** [comment]
I think I fixed it... I hope.

**Comment by mysterious1223** [comment]
I think it should be good now.

File: src/openrct2-ui/windows/TitleCommandEditor.cpp [hunk 1]
```diff
@@ -36,16 +36,16 @@ struct TITLE_COMMAND_ORDER {
 };
 
 static TITLE_COMMAND_ORDER _window_title_command_editor_orders[] = {
-    { TITLE_SCRIPT_LOAD,        STR_TITLE_EDITOR_ACTION_LOAD_SAVE, STR_TITLE_EDITOR_ARGUMENT_SAVEFILE },
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/TitleCommandEditor.cpp:39]: Change the type on line 33. Remove the comment on line 32.

File: src/openrct2/title/TitleSequence.h [hunk 2]
```diff
@@ -58,21 +58,21 @@ struct TitleSequenceParkHandle
     std::unique_ptr<OpenRCT2::IStream> Stream;
 };
 
-enum TITLE_SCRIPT
+enum class TitleScript
```
Comment (duncanspumpkin) [inline src/openrct2/title/TitleSequence.h:61]: Add a size specifier `: uint8_t` change the type on line 22

File: src/openrct2-ui/windows/TitleCommandEditor.cpp (hunk 1 above)
Comment (mysterious1223) [reply src/openrct2-ui/windows/TitleCommandEditor.cpp:39]: Thanks for the feedback. I assume you mean change the `int` to `uint8_t?`

Comment (mysterious1223) [reply src/openrct2-ui/windows/TitleCommandEditor.cpp:39]: I think I see what you mean. Change the type for `command` from 'uint16_t' to `uint8_t`?

Comment (mysterious1223) [reply src/openrct2-ui/windows/TitleCommandEditor.cpp:39]: The struct `TitleCommand` has a member variable `uint16_t Milliseconds`. Should that be adjusted too?
[TitleSequence.h]("src/openrct2/title/TitleSequence.h")

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/TitleCommandEditor.cpp:39]: Command should be type of TitleScript

File: src/openrct2/title/TitleSequence.h [hunk 3]
```diff
@@ -58,21 +58,21 @@ struct TitleSequenceParkHandle
     std::unique_ptr<OpenRCT2::IStream> Stream;
 };
 
-enum TITLE_SCRIPT
+enum class TitleScript : uint8_t
```
Comment (duncanspumpkin) [inline src/openrct2/title/TitleSequence.h:61]: The type on line 22 still needs to be changed to TitleScript

File: src/openrct2/title/TitleSequence.h [hunk 4]
```diff
@@ -58,21 +58,21 @@ struct TitleSequenceParkHandle
...
+    rotate,
+    zoom,
+    follow,
+    restart,
+    load,
+    end,
+    speed,
+    loop,
+    endloop,
+    loadsc,
```
Comment (tupaschoal) [inline src/openrct2/title/TitleSequence.h:75]:
```suggestion
    Undefined = 0xFF,
    Wait = 0,
    Location,
    Rotate,
    Zoom,
    Follow,
    Restart,
    Load,
    End,
    Speed,
    Loop,
    EndLoop,
    LoadSc,
```

We use title case for our enums

**Review by tupaschoal — COMMENTED** [review]
Seems like the rebase wasn't the cleanest, try syncing your fork's develop first then rebasing over it, so that Duncan's commits are dropped

**Comment by mysterious1223** [comment]
Changes made!

File: src/openrct2-ui/windows/TitleCommandEditor.cpp [hunk 5]
```diff
@@ -150,7 +149,7 @@ static int32_t get_command_info_index(int32_t index)
 {
     for (int32_t i = 0; i < static_cast<int32_t>(NUM_COMMANDS); i++)
     {
-        if (_window_title_command_editor_orders[i].command == index)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/TitleCommandEditor.cpp:153]: Index on line 148 should be the TitleScript type. Also rename the variable name to command.

File: src/openrct2-ui/windows/TitleCommandEditor.cpp [hunk 6]
```diff
@@ -160,7 +159,7 @@ static TITLE_COMMAND_ORDER get_command_info(int32_t index)
...
+        if (static_cast<int32_t>(_window_title_command_editor_orders[i].command) == index)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/TitleCommandEditor.cpp:162]: Same as the above function. Line 158 type is TitleScript and name is command

**Comment by duncanspumpkin** [comment]
Just a few more little changes and it's good to go. You should be trying to minimise the use of the static casts where possible.

**Comment by mysterious1223** [comment]
I renamed it to `loc_command` instead. Since `command` is being used as a global variable `TitleCommand command`

**Comment by duncanspumpkin** [comment]
The static global is incorrectly named and should be `_command`. You could also call the functions variable `commandType`

**Comment by mysterious1223** [comment]
updated both. Getting an error during the GitHub checks:
`CI / Linux (Debug, [http, network, OpenGL] disabled) using clang (pull_request)`

**Comment by tupaschoal** [comment]
> updated both. Getting an error during the GitHub checks:
> `CI / Linux (Debug, [http, network, OpenGL] disabled) using clang (pull_request)`

It was an intermittent failure, not related with your change. I've retriggered it and they've all passed now.

**Comment by mysterious1223** [comment]
Awesome. Thanks.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**